### ISSUE TITLE
Android getSharedPreferences file name and mode

### ISSUE THREAD

**Issue by FSpierckel** [jamesmontemagno/SettingsPlugin#17]
I saw that it's not possible to get the sharedPreference with a specified file name. The only way is to use the DefaultSharedPreferences.

It's a possibility for you to add this function? :
 `Context.getSharedPreferences (String name, int mode)`

Thanks in advance

**Comment by codemillmatt** [comment]
So I came here for this exact same reason! And it appears this issue is the Android version of issue #3 ... so there seems to be interest in adding the ability to specify what I'll call a "file name".

But ... as I thought more about this, the more unsure I want to see this implemented. Here's my thought process:

Assuming we want to keep a common abstraction across all the platforms, then a single "file name" would be a parameter on all...
- On Droid the "file name" parameter would be used to specify which physical file the preference resides in.
- On iOS it would identify the "suite name" (assuming initWithSuiteName is used) it lives in - and that's meant for preferences shared across apps & extensions or app groups.
- Since this library is meant to be used in shared code - it doesn't seem right to me that this single "file name" parameter would be used for such different purposes on each platform, it could get confusing.

(Unfortunately I don't know the Windows platforms enough to speak on those, but the point still stands for Droid & iOS.)

Anyway, those are my (unsolicited) thoughts ... 😄 ... I'd be interested in what you think...

**Comment by jamesmontemagno** [comment]
So, it seems like each of the methods needs an additional name associated with it optionally:

```
T GetValueOrDefault<T>(string key, T defaultValue = default(T), string fileName);
```

I am not sure if "fileName" is really want I want to call it, but makes some sense.

For Android Mode there is no need for it as all others besides the default have been deprecated by now.

**Comment by codemillmatt** [comment]
That's what I would think the method signature would look like as well. And I agree with "fileName" not being a great name - but I couldn't think of anything else last night (and nothing better right now either).

But it still kind of bugs me the way iOS would use that parameter isn't analogous to the way Droid would use it. I'm coming from the idea if something is used in shared code, the idea of what it does on each of the platforms is the same - although the implementation is different.

I might be overthinking this whole thing ... it would be useful...

**Comment by jamesmontemagno** [comment]
@codemillmatt Can I get a code review on: https://github.com/jamesmontemagno/SettingsPlugin/tree/feature-filename

Windows I still need to test a bit, but it makes sense to me..

**Comment by codemillmatt** [comment]
iOS and Android look good to me. Couple of nit-picky things:
- In the Android Settings class, change `GetSharedPrefence` to `GetSharedPreference`
- In the iOS Settings class, line 205 - unused commented line

Windows:
I'm confused as to why another `CreateContainer` would get called (line 155) when the key wasn't found during an add of a value. Should line 155 in the Settings class be deleted? Or am I turned around?

**Comment by jamesmontemagno** [comment]
Available for testing on: 2.6.0-beta9

Will apply the changes @codemillmatt has said.
For window those that what was always there... I think you are right though.. not sure why it is there, and all my tests still pass. and doesn't need it according to: https://msdn.microsoft.com/library/windows/apps/windows.storage.applicationdata.localsettings.aspx
